### ISSUE TITLE
Simulate a http client get timeout 

### ISSUE THREAD

**Issue by lneves12** [h2non/gock#64]
Is there a way to simulate a http client timeout? 

I am struggling to find a way how to implement that.

It would be nice to have an example of that on the docs.

Thanks

**Comment by lneves12** [comment]
Finally found a solution:

```golang
type timeoutError struct {
	err     string
	timeout bool
}

func (e *timeoutError) Error() string {
	return e.err
}
func (e *timeoutError) Timeout() bool {
	return e.timeout
}

func (e *timeoutError) Temporary() bool {
	return true
}

var _ = Describe("Testing", func() {
		It("client timeout", func() {
			defer gock.Off()

			gock.New("https://release.memsql.com").
				Get("/production/index/memsqlserver/versions.json").
				ReplyError(&timeoutError{err: "net/http: timeout awaiting response headers", timeout: true})
		})
	})

})

```

I will open a PR later adding this to the examples list

**Comment by h2non** [comment]
You can use the `Delay(duration)` method:
https://godoc.org/github.com/h2non/gock#Response.Delay

Alternatively, you can simulate the native timeout error via:
https://godoc.org/github.com/h2non/gock#Response.SetError

**Comment by lneves12** [comment]
I tried the Dealy(duration) method, but I couldn't simulate the http.client timeout:

```golang
var httpClient = &http.Client{Timeout: 1 * time.Second}
```
I had to use the solution on my second comment. Not sure if it's the best solution though

**Comment by h2non** [comment]
It depends on what type of error you want to simulate. 
If you simply want to test the error handling for a timeout-like situation, that's fine.

**Comment by lneves12** [comment]
yeah, I want to simulate an offline situation, where http.get will timeout. Do you think it's worth to add this as an example to docs? or should we just close the issue?

**Comment by h2non** [comment]
Adding an example file [here](/h2non/gock/tree/master/_examples) should be fine. Feel free to send a PR!
